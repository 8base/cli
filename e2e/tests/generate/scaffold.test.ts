import * as execa from "execa";

import { runner } from "../../runner";

const TEST_APP = "test_app";

beforeEach(async () => {
  await execa.shell(`rm -rf ${TEST_APP}`);
});

it("As a user, I can use help flag for see help information for `generate scaffold` command.", async () => {
  const { stdout } = await runner()("generate", "scaffold", "--help");

  expect(stdout).toMatchSnapshot();
});


it("As a user, I can create some tables.", async () => {
  await runner()("generate", "app", TEST_APP);

  const { stdout: a } = await execa.shell(`pwd`, { cwd: "/Users/ej9x/workspace/cli/test_app"});
  console.log(a);

  const { stdout } = await execa("node", ["/Users/ej9x/workspace/cli", "generate", "scaffold", "Posts"], { cwd: "/Users/ej9x/workspace/cli/test_app" });
  expect(stdout).toMatchSnapshot();

  const { stdout: ls } = await execa.shell(`find ${TEST_APP}/src/routes -mindepth 0 -maxdepth 5 | sort`);
  expect(ls.split("\n").sort()).toMatchSnapshot();

  const { stdout: rootJs } = await execa.shell(`cat ${TEST_APP}/src/Root.js`);
  expect(rootJs).toMatchSnapshot();

  await execa.shell(`rm -rf ${TEST_APP}`);
});

