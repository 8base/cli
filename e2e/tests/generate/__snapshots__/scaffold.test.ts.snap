// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`As a user, I can create some tables. 1`] = `
"error: Can't find a '.8base.yml' file. You should be in the project root directory to exec this command. 
 Time: 100 ms."
`;

exports[`As a user, I can create some tables. 2`] = `
Array [
  "test_app/src/routes",
  "test_app/src/routes/auth",
  "test_app/src/routes/auth/index.js",
  "test_app/src/routes/auth/routes",
  "test_app/src/routes/auth/routes/callback",
  "test_app/src/routes/auth/routes/callback/index.js",
  "test_app/src/routes/auth/routes/root",
  "test_app/src/routes/auth/routes/root/index.js",
]
`;

exports[`As a user, I can create some tables. 3`] = `
"import React from 'react';
import { Redirect, Route, Switch } from 'react-router-dom';

import { ProtectedRoute } from './shared/components';
import { MainPlate, ContentPlate, Nav } from './components';
import { Auth } from './routes/auth';
/** __APP_PAGES_IMPORTS__ */

export const Root = () => (
  <Switch>
    <Route path=\\"/auth\\" component={Auth} />
    <Route>
      <MainPlate>
        <Nav.Plate color=\\"BLUE\\">
          {/** __APP_ROUTE_LINKS__ */}
        </Nav.Plate>
        <ContentPlate>
          <Switch>
            {/** __APP_ROUTES__ */}
        </Switch>
        </ContentPlate>
      </MainPlate>
    </Route>
  </Switch>
);"
`;

exports[`As a user, I can use help flag for see help information for \`generate scaffold\` command. 1`] = `
"8base generate scaffold [TABLE_NAME] [OPTIONS]

Options:
  --debug, -d  Turn on debug logs                                      [boolean]
  --help       Show help                                               [boolean]
  --depth      Depth of the generated query                [number] [default: 1]"
`;
